bitcoin/test/fuzz/miniscript_decode.cpp

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2019-2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <core_io.h>
+#include <hash.h>
+#include <script/miniscript.h>
+#include <script/script.h>
+#include <span.h>
+#include <test/fuzz/FuzzedDataProvider.h>
+#include <test/fuzz/fuzz.h>
+#include <test/fuzz/util.h>
+#include <util/strencodings.h>
+
+
+#include <optional>
+
+/** A class encapulating conversion routines for CPubKey. Since we can't have access to a
+ * hash->pubkey mapping, and since fuzzing the conversion itself here would be useless
+ * computation we use a static dummy public key and its hash. */
+struct MockedKeyConverter {
+    typedef CPubKey Key;
+
+    std::string dummy_key_hex;
+    std::vector<unsigned char> dummy_key_ser;
+    std::vector<unsigned char> dummy_key_hash_ser;
+    CPubKey dummy_key;
+    uint160 dummy_key_hash;
+
+    MockedKeyConverter() :
+        dummy_key_hex("02d7924d4f7d43ea965a465ae3095ff41131e5946f3c85f79e44adbcf8e27e080e"),
+        dummy_key_ser(ParseHex(dummy_key_hex)),
+        dummy_key_hash_ser(ParseHex("9fc5dbe5efdce10374a4dd4053c93af540211718")),
+        dummy_key(dummy_key_ser),
+        dummy_key_hash(dummy_key_hash_ser) {}
+
+    bool ToString(const CPubKey& key, std::string& ret) const { ret = dummy_key_hex; return true; }
+
+    std::vector<unsigned char> ToPKBytes(const CPubKey& key) const { return dummy_key_ser; }
```

Comment:
You can return a const reference here to avoid a copy (here and below).

Reply:
We are now returning an actual value so this can't be done.